newIDE/app/public/external/piskel/piskel-main.js

```diff
@@ -328,40 +194,221 @@ ipcRenderer.on('piskel-load-animation', (event, receivedOptions) => {
+    receivedPiskelData,
+    piskel => {
+      piskelController.setPiskel(piskel);
+
+      // set piskel frame paths to their piskel data counterpart - on all layers
+      piskelController.getLayers().forEach(layer => {
+        layer.getFrames().forEach((frame, index) => {
+          frame.originalPath = editorDataPaths[index];
+          frame.originalIndex = index;
+        });
+      });
+
+      // Compare the imported frames - so as to make the layered Piskel Document
+      // the same as the changes done in Gdevelop without flattening any layers
+      let flattenedImagePaths = [];
+      piskelOptions.resources.forEach((resource, frameIndex) => {
+        const flattenedFramePath = path.normalize(resource.resourcePath);
+        flattenedImagePaths.push(flattenedFramePath);
+
+        // Import any frames that were added in Gdevelop
+        if (!editorDataPaths.includes(flattenedFramePath)) {
+          pskl.utils.BlobUtils.dataToBlob(
+            readBase64ImageFile(flattenedFramePath),
+            'image/png',
+            imageBlob => {
+              pskl.utils.FileUtils.readImageFile(imageBlob, image => {
+                piskelController.setCurrentFrameIndex(
+                  piskelController.getFrameCount() - 1
+                );
+
+                piskelController.addFrameAtCurrentIndex();
+                piskelController.selectNextFrame();
+                const currentFrameObj = piskelController.getCurrentFrame();
+                pskl.utils.FrameUtils.addImageToFrame(
+                  currentFrameObj,
+                  image,
+                  0,
+                  0
+                );
+                pskl.tools.transform.TransformUtils.center(currentFrameObj);
+
+                piskelController.getLayers().forEach(layer => {
+                  layer.moveFrame(
+                    piskelController.getCurrentFrameIndex(),
+                    frameIndex
+                  );
+                  layer.getFrameAt(frameIndex).originalIndex = frameIndex;
+                  layer.getFrameAt(
+                    frameIndex
+                  ).originalPath = flattenedFramePath;
+                });
+              });
+            }
+          );
+        }
+      });
+
+      // Remove any frames that were removed in GD
+      let framesToDelete = [];
+      piskelController
+        .getLayers()[0]
+        .getFrames()
+        .forEach(frame => {
+          // frame was in meta, but is not in GDframes, if so remove it
+          if (
+            editorDataPaths.includes(frame.originalPath) &&
+            !flattenedImagePaths.includes(frame.originalPath)
+          ) {
+            framesToDelete.push(frame);
+          }
+        });
+      framesToDelete.forEach(frameToDelete => {
+        const layer = piskelController.getLayers()[0];
+        const removeFrameIndex = layer.getFrames().indexOf(frameToDelete);
+        if (removeFrameIndex !== -1) {
+          if (piskelController.getFrameCount() === 1) {
```

Comment:
I'm curious, why do we need a special case for frame count being 1? :)
I'm not sure to understand the comment

Reply:
The loop iterates through piskel's frames and deletes them if they were removed in gdevelop. 
I found out that if all frames get deleted here, that results in piskel spilling a lot of errors and locking out the user. So I added this case to always keep them at 1 by inserting an empty one when the critical number is reached

Reply:
You can try this out by commenting this if case, then in piskel save multiple frames with layers. In gdevelop delete all frames and add a couple of new ones, then open piskel

Reply:
I see, I guess Piskel is not handling having no frames at all! Thanks for the explanation

Reply:
would you like me to update the comment or look for another way around it? :)

Reply:
I updated the comment to make this clearer :)

Reply:
Thanks! :)